### DISCUSSION TITLE
Separate examples, possibly tests, into their own repo

### DISCUSSION THREAD

**Discussion by RobertPincus** [earth-system-radiation/rte-rrtmgp#191] [Ideas]
@Chiil @menzel-gfdl As we expand the universe of gas optics treatments it would be nice to reuse as much of the example and testing codes as possible. A proposal: we separate the Fortran codes currently in examples/ and tests/ into their own repo and generalize them to work with any gas optics object. We write small modules for each specific gas optics implementation that define the specific gas optics type and initialize it with data. At compile time one links the general codes with the specific gas optics type. 

Continuous integration would check out, build, and run the codes in the example-n-testing repo. 

Comments? Planning to do this in January if it's ok.

**Comment by RobertPincus** [comment]
@vectorflux Curious if you have any thoughts on this too, esp. if it complicates CI that we're hosting.

**Comment by RobertPincus** [comment]
@skosukhin Also interested to know if you have an opinion on this question

**Comment by Chiil** [comment]
Happy to talk soon about organization of code and tests!

**Comment by RobertPincus** [comment]
After some discussion we've agreed to generalize the examples and tests to allow for different gas optics implementations, but not to remove the examples and tests from the main repo.
